astrodb_utils/spectra.py

```diff
@@ -158,3 +170,325 @@ def _plot_spectrum(spectrum):
+    logger.debug(f"Parsing obs_date: {obs_date}")
+    if obs_date is not None and isinstance(obs_date, str):
+        parsed_date = check_obs_date(obs_date, raise_error=raise_error)
+    elif isinstance(obs_date, datetime.datetime):
+        parsed_date = obs_date
+
+    if obs_date is None or (parsed_date is None and raise_error is False):
+        msg = f"Observation date is not valid: {obs_date}"
+        flags["message"] = msg
+        if raise_error:
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return flags
+
+    # Get source name as it appears in the database
+    db_name = find_source_in_db(db, source)
+    logger.debug(f"Found db_name: {db_name} for source: {source}")
+    if len(db_name) == 1:
+        db_name = db_name[0]
+    else:
+        msg = f"No unique source match for {source} in the database. Found {db_name}."
+        flags["message"] = msg
+        if raise_error:
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return flags
+
+    # Check if spectrum is a duplicate
+    matches = find_spectra(
+        db,
+        db_name,
+        reference=reference,
+        obs_date=parsed_date,
+        telescope=telescope,
+        instrument=instrument,
+        mode=mode,
+    )
+    if len(matches) > 0:
+        msg = f"Skipping suspected duplicate measurement: {source}"
+        msg2 = f"{matches} {instrument, mode, parsed_date, reference, spectrum}"
+        logger.debug(msg2)
+        flags["message"] = msg
+        # exit_function(msg, raise_error=raise_error)
+        if raise_error:
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return flags
+
+    # Make sure reference is provided and is in the Publications table
+    if reference is None:
+        msg = "Reference is required."
+        flags["message"] = msg
+        exit_function(msg, raise_error=raise_error, return_value=flags)
+        return flags
+    else:
+        reference_db = get_db_publication(db, reference, raise_error=raise_error)
+        if reference_db is None:
+            flags["message"] = f"Reference not found in database: {reference}."
+            return flags
+
+    # Check if regime is provided and is in the Regimes table         
+    regime = get_db_regime(db, regime, raise_error=raise_error)
+    if regime is None:
+        msg = f"Regime not found in database: {regime}."
+        flags["message"] = msg
+        if raise_error:
+            raise AstroDBError(msg)
+        else:
+            logger.warning(msg)
+            return flags
+
+    # Find the right instrument-mode-telescope in the database
+    instrument, mode, telescope = get_db_instrument(
+        db,
+        instrument=instrument,
+        mode=mode,
+        telescope=telescope,
+    )
+
+    # Check if spectrum file(s) are accessible
+    check_spectrum_accessible(spectrum)
```

Comment:
`check_spectrum_accessible` returns True/False, in addition to throwing errors. Do you not want to capture if it's True/False like you do below for plottable? 

I think if you don't capture the output you might get random print statements with True/False when you execute this.

Reply:
I'm a bit confused by this comment.  `check_spectrum_accessible` only raises errors if there's **no** internet connection.  Otherwise, it _logs_ an error, but returns false.